codex/blockexchange/engine/pendingblocks.nim

```diff
@@ -39,12 +39,12 @@ proc getWantHandle*(
 
   if cid notin p.blocks:
      p.blocks[cid] = newFuture[Block]().wait(timeout)
-     trace "Adding pending future for block", cid
+     trace "Adding pending future for block", cid = $cid
 
   try:
     return await p.blocks[cid]
   except CancelledError as exc:
-    trace "Blocks cancelled", exc = exc.msg, cid
+    trace "Blocks cancelled", exc = exc.msg, cid = $cid
```

Comment:
For a reason I don't understand yet, passing `cid` to chronicle's trace started causing compilation errors, with json_serialization complaining about MultiCodec.

I tried bumping the serialization libraries and chronicles, but it didn't solve the problem. As a quick workaround, I introduced changes such as `cid = $cid` throughout the codebase.

Reply:
This requires chronicles to know how to serialize it properly, it would be nice if we could add a `formatIt()` for `Cid` in libp2p, to avoid re-defining it everywhere.

Reply:
It was working previously, though. It only started being a problem owing to one/more of the deps that I needed to bump to get the tests compiling with nim-libp2p#unstable. When I got close to submitting the changes as a draft PR, I tried `make exec` and ran into this problem.

Reply:
Feel free to open PRs in libp2p :)

Reply:
@michaelsbradleyjr if you look through the codebase, there should be plenty of examples of how to add a custom formatter, just search for `formatIt`.

Actually, I think adding the formatter to codex directly is probably the best way to do it, after all libp2p shouldn't care about that.